CombinatorialGames/Game/Birthday.lean

```diff
@@ -224,7 +231,174 @@ instance small_setOf_birthday_lt (o : NatOrdinal.{u}) : Small.{u} {x // birthday
   apply @small_subset _ _ _ _ (small_setOf_birthday_le o)
   exact fun x (hx : x.birthday < _) ↦ le_of_lt hx
 
--- TODO: short game iff finite birthday
+/-- The finset of all games with birthday ≤ n. -/
+noncomputable def birthdayFinset : ℕ → Finset IGame.{u}
+  | 0 => {0}
+  | n + 1 => ((birthdayFinset n).powerset ×ˢ (birthdayFinset n).powerset).map
+    ⟨fun ⟨a, b⟩ => {a | b}ᴵ, fun a b hab => by aesop⟩
+
+theorem birthdayFinset_mem_succ_iff {x : IGame} {n : ℕ} :
+  x ∈ birthdayFinset (n + 1) ↔
+    ∃ l r, (l ⊆ birthdayFinset n ∧ r ⊆ birthdayFinset n) ∧ {↑l | ↑r}ᴵ = x := by
+  simp [birthdayFinset]
+
+@[simp] theorem birthdayFinset_zero : birthdayFinset 0 = {0} := rfl
+
+@[simp] theorem card_birthdayFinset_one : (birthdayFinset 1).card = 4 := by simp [birthdayFinset]
```

Comment:
Something interesting is that we actually have names for all the games of birthday ≤1. In fact, check this out:
```lean 4
@[simp]
theorem Set.empty_ne_singleton {α : Type*} (a : α) : ∅ ≠ ({a} : Set α) :=
  (Set.singleton_ne_empty a).symm

theorem birthdayFinset_one :
    birthdayFinset 1 = ⟨[0, 1, -1, ⋆], by aesop (add simp [IGame.ext_iff])⟩ := by
  ext
  rw [birthdayFinset_mem_succ_iff]
  aesop (add simp [IGame.ext_iff])
```
You can add it into the PR, or we can do this in a separate PR, however you prefer.

Reply:
I can add it into here :+1: